modules/components/tag-input/tag-input.ts

```diff
@@ -509,11 +509,20 @@ export class TagInputComponent extends TagInputAccessor implements OnInit, After
      * @name appendTag
      * @param tag {TagModel}
      */
-    public appendTag = (tag: TagModel): void => {
-        this.items = [
-            ...this.items,
-            this.modelAsStrings ? tag[this.identifyBy] : tag
-        ];
+    public appendTag = (tag: TagModel, index = this.items.length): void => {
+        const items = [];
+
+        for (let i = 0; i <= this.items.length; i++) {
+            if (i < index) {
+                items.push(this.items[i]);
+            } else if (i === index) {
+                items.push(this.modelAsStrings ? tag[this.identifyBy] : tag);
+            } else {
+                items.push(this.items[i - 1]);
+            }
+        }
```

Comment:
can you not reuse 
```
if (index === undefined) {		
             this.appendTag(tag);		
        } else {		
            const items = this.items;		
            this.items = [...items.slice(0, index), tag, ...items.slice(index, items.length)];		
        }
```

or does it do something wrong? I'm asking as this the latter seems more concise and easy to understand than a for loop